dns/dns_updater.go

```diff
@@ -264,31 +303,39 @@ func (u *updater) createChanges(hostToIngress hostToIngress,
 		}
 	}
 
-	for _, recordSet := range originalRecords {
-		if _, contains := hostToIngress[*recordSet.Name]; !contains {
-			changes = append(changes, newChange(
-				"DELETE",
-				*recordSet.Name,
-				*recordSet.AliasTarget.DNSName,
-				*recordSet.AliasTarget.HostedZoneId))
+	for _, rec := range originalRecords {
+		if _, contains := hostToIngress[rec.name]; !contains {
+			changes = append(changes, newChange("DELETE", rec.name, rec.pointsTo, rec.aliasHostedZone))
 		}
 	}
 
 	return changes, skipped
 }
 
 func newChange(action string, host string, targetElbDNSName string, targetElbHostedZoneID string) *route53.Change {
-	return &route53.Change{
-		Action: aws.String(action),
-		ResourceRecordSet: &route53.ResourceRecordSet{
-			Name: aws.String(host),
-			Type: aws.String("A"),
-			AliasTarget: &route53.AliasTarget{
-				DNSName:      aws.String(targetElbDNSName),
-				HostedZoneId: aws.String(targetElbHostedZoneID),
-				// disable this since we only point to a single load balancer
-				EvaluateTargetHealth: aws.Bool(false),
+	set := &route53.ResourceRecordSet{
+		Name: aws.String(host),
+	}
+
+	if targetElbHostedZoneID != "" {
+		set.Type = aws.String("A")
+		set.AliasTarget = &route53.AliasTarget{
+			DNSName:      aws.String(targetElbDNSName),
+			HostedZoneId: aws.String(targetElbHostedZoneID),
+			// disable this since we only point to a single load balancer
+			EvaluateTargetHealth: aws.Bool(false),
+		}
+	} else {
+		set.Type = aws.String("CNAME")
+		set.ResourceRecords = []*route53.ResourceRecord{
+			{
+				Value: aws.String(targetElbDNSName),
 			},
-		},
+		}
+	}
+
+	return &route53.Change{
+		Action:            aws.String(action),
+		ResourceRecordSet: set,
 	}
 }
```

Comment:
Since there is a large amount of conditional logic in here now, I wonder if we should split up this file into two implementations. One for the ELB/ALBs, and one for the static hostnames. I think it'd make it easier to understand as well.